### DISCUSSION TITLE
Specify Shell Type

### DISCUSSION THREAD

**Discussion by ghost** [github-linguist/linguist#5362] [General]
## Although it is a great thing that the Shell language has been included in `languages.yml`, it can be difficult to find out what shell(s) a repository is meant for. 

After all, each different type has their own set of commands that they support.

There is a caveat or two with this enhancement request, though. 
A lot of the files are not labeled by the extension, but rather by the [_shebang_](https://en.wikipedia.org/wiki/Shebang_(Unix)). So I am thinking that there will have to be a way to check what type it is (unless the [interpreter](https://github.com/github/linguist/blob/e6860ab64b463e022294b0f71d468d5243687027/lib/linguist/languages.yml#L5377) already accomplishes that for us). 
The other concern is simply letting users know the specific type while also conveying that it is still a shell script. Maybe different names, but with the same color? Maybe even `Shell - Bash` or `Shell - Zsh` instead of `Shell`, and keep the same color that the Shell language already uses.

Besides the suggestion above, it could be done easily if there is a way to make some kind of customized name with a `.gitattributes` file like `linguist-language`.

**Comment by TerabyteTB** [comment]
I think sometimes Ruby uses shebangs too, but I'm not sure.
I also found a `shebang.rb` in the repository.

**Comment by ghost** [comment]
I did see the [script](https://github.com/github/linguishebang.rbst/blob/master/lib/linguist/shebang.rb), but it looks like it merely [passes its findings](https://github.com/github/linguist/blob/32ec19c013a7f81ffaeead25e6e8f9668c7ed574/lib/linguist/shebang.rb#L14) on to other files. 
What is required is the file(s) that interact with its findings.

**Comment by TerabyteTB** [comment]
oh, I only know a tiny bit about Ruby

**Comment by ghost** [comment]
_No worries!_
Fortunately, as we know, most languages have a similar syntax for us to get a gist of how they work.

Right now we only need to see how the detection is used. If it passes the shebang type as "'shell'", then it will respond accordingly. This is where we come in to find where the result(s) get processed.

**Comment by TerabyteTB** [comment]
I think it's [here](https://github.com/github/linguist/blob/32ec19c013a7f81ffaeead25e6e8f9668c7ed574/lib/linguist.rb#L32).

(Sorry if it's kind of obvious)

**Comment by ghost** [comment]
Thanks for that!

**Comment by ghost** [comment]
However we go about this, it looks like one would have to modify [`languages.yml`](https://github.com/github/linguist/blob/32ec19c013a7f81ffaeead25e6e8f9668c7ed574/lib/linguist/languages.yml#L5346).

**Comment by TerabyteTB** [comment]
Huh. Maybe someone can add a `shell-type` in the Shell section and put whatever in there into a variable of some sort, change it into a name, and display it after the `Shell` name.

**Comment by ghost** [comment]
Right.

**Comment by TerabyteTB** [comment]
I would really like to know how does `grammers.yml` work

**Comment by ghost** [comment]
[It is a collection of files that define the coloring of words when they're uploaded to GitHub.](https://github.com/github/linguist/blob/master/vendor/README.md#grammar-index)

Were you thinking of something of it that relates to this issue?

**Reply by TerabyteTB** [reply]
No, I’m just curious about what it does and whether it’s related. Apparently not.

**Comment by chapodaddy42** [comment]
> `` #5298

**Comment by chapodaddy42** [comment]
Agree with you

**Reply by TerabyteTB** [reply]
What?
